### DISCUSSION TITLE
What am I doing wrong? "TypeError: Cannot use 'in' operator to search for 'undefined' in null", and I am using the exact path that the documentation said.

### DISCUSSION THREAD

**Discussion by MhmdNdri** [nextauthjs/next-auth#1648] [Help]
**Your question**
What am I doing wrong? I am getting this error during runtime
TypeError: Cannot use 'in' operator to search for 'undefined' in null.

**What are you trying to do**
Trying to setup authentication for my Project.

**Comment by balazsorban44** [comment]
please add a complete reproduction

**Reply by MhmdNdri** [reply]
**Reproduction**
When I clicked on the sign-in button, it happens every time.

**Reply by balazsorban44** [reply]
we will need more than that. maybe if your repo is public, could you link to it? if not, at least please share your nextauth config, the way you are trying to signIn in the frontend, and your NEXTAUTH_URL env variable value

**Reply by MhmdNdri** [reply]
here is my repository link : https://github.com/MhmdNdri/Next-Access
I don't set any NEXTAUTH-URL env variable for testing on my local host and I think it's ok.

**Reply by MhmdNdri** [reply]
> we will need more than that. maybe if your repo is public, could you link to it? if not, at least please share your nextauth config, the way you are trying to signIn in the frontend, and your NEXTAUTH_URL env variable value

Did you find any bug in my configuration or my code that help me solve this issue?

**Reply by balazsorban44** [reply]
So after cloning and trying to run your project locally, I hit an error that does not originate from `next-auth`, but rather how some other mechanism overrides the base path. for `next-auth` it is expected that the `/api/auth/[action]` path will point to the `/pages/api/auth/[...nextauth].js` file, but in your case, it is only available under `/pwa-sample/api/auth/[...nextauth].js`. This creates 404 responses that in return cause the issue you describe. I saw that your project had a hello.js API file as well, and I confirmed that it also only works from under /pwa-sample/hello, instead of /hello. Any way of letting next-auth use the /api/auth without the /pwa-sample part? Our client-side URL parser is hardcoded to that URL at:
https://github.com/nextauthjs/next-auth/blob/5bc8f8b986f84c0e42a71cc2c87f64b78cc8554c/src/lib/parse-url.js#L11

cc: @iaincollins is this intended? Do we offer a way to resolve this? The reason is probably that the env variable is not prepended with `NEXT_PUBLIC_` and so Next.js just inlines `undefined` at build-time for the client-side URL
Overriding the variable name in `node_modules/next-auth/dist/client.js` to `NEXT_PUBLIC_NEXTAUTH_URL` worked for me locally. If we wanted to change that though, it would probably be a breaking change.


@MhmdNdri to reiterate, currently, the only way I can see for you to fix this is to somehow tell Next.js that the /api/auth URL should point to the /pages/api/auth/[...nextauth].js file, without the /pwa-sample part
